gtsam/hybrid/HybridConditional.cpp

```diff
@@ -102,7 +102,20 @@ void HybridConditional::print(const std::string &s,
 /* ************************************************************************ */
 bool HybridConditional::equals(const HybridFactor &other, double tol) const {
   const This *e = dynamic_cast<const This *>(&other);
-  return e != nullptr && BaseFactor::equals(*e, tol);
+  if (e == nullptr) return false;
```

Comment:
I understand this change now, but unfortunately this is breaking serialization since lots of Discrete factors and other hybrid factors/conditionals don't support serialization.